### DISCUSSION TITLE
Learning Home Assistant -

### DISCUSSION THREAD

**Discussion by eralbright** [jeroenterheerdt/HAsmartirrigation#352] [General]
So I've been doing Home Automation since the 90's but just moving to Home Assistant.  I have used your smart irrigation for a lot of learning and having fun with it.  Thanks...  This is (not your intent for the program) what I put together and it works.  Again it was not your intent of the program but my learning by using it.

`alias: Foundation Watering
description: >-
  Drip watering my foundation whenever the need for water is > 0.25 inches and
  stop when the need is < 0.1 inches.  Water in 5 minute intervals with a 3
  minute delay between intervals. Still to implement: check for rain and don't
  run if raining.
trigger:
  - platform: state
    entity_id:
      - calendar.watering_schedule_foundation
    from: "off"
    to: "on"
condition:
  - condition: state
    state: "on"
    entity_id: calendar.watering_schedule_foundation
  - condition: template
    value_template: >-
      {{state_attr('sensor.smart_irrigation_zone_7_foundation','bucket') <
      -0.25}}
  - condition: numeric_state
    entity_id: sensor.friscoweather_temperature
    above: 48
action:
  - service: smart_irrigation.calculate_all_zones
    data: {}
  - repeat:
      while:
        - condition: template
          value_template: >-
            {{state_attr('sensor.smart_irrigation_zone_7_foundation','bucket') |
            float < -0.1 }}
      sequence:
        - service: smart_irrigation.set_bucket
          data:
            new_bucket_value: >-
              {{state_attr('sensor.smart_irrigation_zone_7_foundation','bucket')
              | float + 0.025 }}
          target:
            entity_id: sensor.smart_irrigation_zone_7_foundation
        - type: turn_on
          device_id: ef8bd15cd34ed99ac0d4f982f397e4a0
          entity_id: 6110ebb0000cd9fb5d64cbbedc01e648
          domain: switch
        - delay: "00:05:00"
        - type: turn_off
          device_id: ef8bd15cd34ed99ac0d4f982f397e4a0
          entity_id: 6110ebb0000cd9fb5d64cbbedc01e648
          domain: switch
        - delay: "00:03:00"
        - if:
            - condition: state
              entity_id: calendar.watering_schedule_foundation
              state: "off"
          then:
            - stop: Outside approved watering times
mode: single`

**Comment by jeroenterheerdt** [comment]
that's cool - but why are you watering your foundation? :)

**Reply by eralbright** [reply]
In Texas, the soil is clay and moves a lot based on how wet it is which can cause the house to move and even become tipped.  To minimize this, it is common to add water to about 1 foot perimeter of the house so the doesn't move.  This automation runs the rest of the irrigation as well but is a lot longer checking for delay if no other zone is watered or runs them in sequence if multiple zones need water.  If it is a spray zone, it also checks wind speed.

**Reply by jeroenterheerdt** [reply]
Ha, cool thanks for explaining. I learned something today. Thank you.
